src/mlpack/methods/bayesian_ridge/bayesian_ridge.hpp

```diff
@@ -0,0 +1,298 @@
+ *
+ * More details and description in :
+ * Christopher Bishop (2006), Pattern Recognition and Machine Learning.
+ * David J.C MacKay (1991), Bayesian Interpolation, Computation and Neural
+ * systems.
+ 
+ * Model optimization is automatic and does not require cross validation
+ * procedure to be optimized.
+ *
+ * @code
+ * @article{MacKay91bayesianinterpolation,
+ *   author = {David J.C. MacKay},
+ *   title = {Bayesian Interpolation},
+ *   journal = {NEURAL COMPUTATION},
+ *   year = {1991},
+ *   volume = {4},
+ *   pages = {415--447}
+ * }
+ * @endcode
+ *
+ * @book{Bishop:2006:PRM:1162264,
+ *   author = {Bishop, Christopher M.},
+ *   title = {Pattern Recognition and Machine Learning (Information Science 
+ *            and Statistics)},
+ *   chapter = {3}
+ *   year = {2006},
+ *   isbn = {0387310738},
+ *   publisher = {Springer-Verlag},
+ *   address = {Berlin, Heidelberg},
+ * } 
+ * @encode
+ *   
+ * Example of use:
+ *
+ * @code
+ * arma::mat Xtrain; // Train data matrix. Column-major.
+ * arma::rowvec ytrain; // Train target values.
+ 
+ * // Train the model. Regularization strength is optimally tunned with the
+ * // training data alone by applying the Train method.
+ * BayesianRidge estimator(); // Instanciate the estimator with default option.
+ * estimator.Train(Xtrain, ytrain);
+ 
+ * // Prediction on test points.
+ * arma::mat Xtest; // Test data matrix. Column-major.
+ * arma::rowvec predictions;
+ 
+ * estimator.Predict(Xtest, prediction);
+ 
+ * arma::rowvec ytest; // Test target values.
+ * estimator.Rmse(Xtest, ytest); // Evaluate using the RMSE score.
+ 
+ * // Compute the standard deviations of the predictions.
+ * arma::rowvec stds;
+ * estimator.Predict(Xtest, responses, stds)
+ * @endcode
+ */
+class BayesianRidge
+{
+ public:
+  /**
+   * Set the parameters of Bayesian Ridge regression object. The
+   *    regulariation parameter is automaticaly set to its optimal value by
+   *    maximmization of the marginal likelihood.
+   *
+   * @param centerData Whether or not center the data according to the
+   *    examples.
+   * @param scaleData Whether or to scale the data according to the
+   *    standard deviation of each feature.
+   * @param nIterMax Maximum number of iterations for convergency.
+   * @param tol Level from which the solution is considered sufficientlly 
+   *    stable.  
+   **/
+  BayesianRidge(const bool centerData = true,
+                const bool scaleData = false,
+                const int nIterMax = 50,
+                const double tol = 1e-4);
+
+  /**
+   * Run BayesianRidge regression. The input matrix (like all mlpack matrices)
+   * should be
+   * column-major -- each column is an observation and each row is a dimension.
+   * 
+   * @param data Column-major input data
+   * @param responses A vector of targets.
+   * @return score. Root Mean Square Error. Equal to -1 if two feature vectors 
+   *    or more are colinear.
```

Comment:
I'd suggest instead throwing an exception if there is a problem.  That would better match other mlpack methods.

Reply:
Now std::runtime_error is thrown.